android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/temporary/TemporaryDownloadFeature.kt

```diff
@@ -63,51 +47,23 @@ private const val OPERATION_TIMEOUT_MILLIS = 1000L
 internal var cacheDirName = "mozac_share_cache"
 
 /**
- * [Middleware] implementation for sharing online resources.
- *
- * This will intercept only [ShareInternetResourceAction] [BrowserAction]s.
- *
- * Following which it will transparently
- *  - download internet resources while respecting the private mode related to cookies handling
- *  - temporarily cache the downloaded resources
- *  - automatically open the platform app chooser to share the cached files with other installed Android apps.
- *
- * with a 1 second timeout to ensure a smooth UX.
+ * Base class for downloading resources from the internet and storing them in a temporary cache.
  *
- * To finish the process in this small timeframe the feature is recommended to be used only for images.
- *
- *  @param context Android context used for various platform interactions
- *  @param httpClient Client used for downloading internet resources
- *  @param store a reference to the application's [BrowserStore]
- * param Coroutine dispatcher used for the cleanup of old cached files. Defaults to IO.
-*/
-class ShareDownloadFeature(
+ *  @property context Android context used for various platform interactions.
+ *  @property httpClient Client used for downloading internet resources.
+ *  @param cleanupCacheCoroutineDispatcher Coroutine dispatcher used for the cleanup of old
+ *  cached files. Defaults to IO.
+ */
+abstract class TemporaryDownloadFeature(
     private val context: Context,
     private val httpClient: Client,
-    private val store: BrowserStore,
-    private val tabId: String?,
     cleanupCacheCoroutineDispatcher: CoroutineDispatcher = IO,
 ) : LifecycleAwareFeature {
-    private val logger = Logger("ShareDownloadMiddleware")
 
-    @VisibleForTesting
-    internal var scope: CoroutineScope? = null
-
-    override fun start() {
-        scope = store.flowScoped { flow ->
-            flow.mapNotNull { state -> state.findTabOrCustomTabOrSelectedTab(tabId) }
-                .ifChanged { it.content.share }
-                .collect { state ->
-                    state.content.share?.let { shareState ->
-                        logger.debug("Starting the sharing process")
-                        startSharing(shareState)
-
-                        // This is a fire and forget action, not something that we want lingering the tab state.
-                        store.dispatch(ShareInternetResourceAction.ConsumeShareAction(state.id))
-                    }
-                }
-        }
-    }
+    val logger = Logger("TemporaryDownloadMiddleware")
```

Comment:
nit: Leftover from previous implementations: The class is named `TemporaryDownloadFeature`  but the log tag is `TemporaryDownloadMiddleware`.